### PR TITLE
Add pkgname-user-in-source

### PR DESCRIPTION
**Pull request by Ikke**

### CODE REVIEW

**Comment by Ikke** [comment]
Not sure if we want to make this an official rule, but I saw it as a use case trying to implement a check myself.

**Comment by maxice8** [comment]
I actually do use $pkgname and $pkgver a lot in source=

**Comment by Ikke** [comment]
This is purely `$pkgname` in the source url itself, and about decoupling the alpine package name from the upstream name.

```sh
source=$pkgname-$pkver::https://domain.com/package-$pkgver.tar.gz
```
would not give a warning.

**Comment by maxice8** [comment]
> This is purely $pkgname in the source url itself

That is what i use quite a lot.

> and about decoupling the alpine package name from the upstream name.

Makes some sense. Is it alpine policy ?

**Comment by maxice8** [comment]
It is missing an addition to alint.5

**Comment by Ikke** [comment]
> It is missing an addition to alint.5

Added

> > and about decoupling the alpine package name from the upstream name.
> 
> Makes some sense. Is it alpine policy ?

Just had a discussion with ncopa in #alpine-devel, and he tends to agree, $pkgname should be used sparingly, rather than using it for every occurance for the upstream name.

I guess this should be documented in the codestyle as well.

File: alint.5.scd [hunk 1]
```diff
@@ -312,3 +312,11 @@ Severity: Serious, Certainty: Certain
 A variable declaration containing only integers should not be quoted.
 
 Severity: Minor,  Certainty: Certain
+
+## pkgname-used-in-source [AL29]
+
+"$pkgname" is used in the source url. This tightly couples the pkgname to the
+upstream name, which makes it harder to rename packages or create specialized /
+variants of packages.
+
```
Comment (maxice8) [inline alint.5.scd]: Please add a second paragraph with a recommendation to use the full written-out name instead.

**Comment by maxice8** [comment]
> I guess this should be documented in the codestyle as well.

@oliv3r this might be of interest

**Comment by Ikke** [comment]
> > I guess this should be documented in the codestyle as well.
> 
> @oliv3r this might be of interest

I was already working on a pull request for this.

File: alint.5.scd (hunk 1 above)
Comment (Ikke) [reply alint.5.scd]: Added

**Comment by oliv3r** [comment]
> > I guess this should be documented in the codestyle as well.
> 
> @oliv3r this might be of interest

Thanks, I think this is covered in alpine/aports!12 but can discuss it in more detail there? "For example, do not use it in fields that couple things like URL to the pkgname and installed files.

**Pull request merged**